package.json

```diff
@@ -42,21 +42,22 @@
     "postinstall": "patch-package",
     "prebuild": "rimraf build/ build-bin/",
     "build": "npm run build:src && npm run catch-uncommitted",
-    "build:src": "npm run prettify && npm run lint && npm run build:fast && npm run build:doc",
+    "build:src": "npm run prettify && npm run lint && npm run build:automation && npm run build:fast && npm run build:doc",
     "build:fast": "gulp build && tsc",
-    "build:doc": "mkdirp doc/ && ts-node --type-check -P automation/tsconfig.json automation/capitanodoc/index.ts > doc/cli.markdown",
-    "build:standalone": "ts-node --type-check -P automation/tsconfig.json automation/run.ts build:standalone",
-    "build:installer": "ts-node --type-check -P automation/tsconfig.json automation/run.ts build:installer",
+    "build:automation": "tsc -P automation/tsconfig.json --noEmit",
+    "build:doc": "mkdirp doc/ && ts-node --transpile-only -P automation/tsconfig.json automation/capitanodoc/index.ts > doc/cli.markdown",
+    "build:standalone": "ts-node --transpile-only -P automation/tsconfig.json automation/run.ts build:standalone",
+    "build:installer": "ts-node --transpile-only -P automation/tsconfig.json automation/run.ts build:installer",
     "package": "npm run build:fast && npm run build:standalone && npm run build:installer",
-    "release": "ts-node --type-check -P automation/tsconfig.json automation/run.ts release",
+    "release": "ts-node --transpile-only -P automation/tsconfig.json automation/run.ts release",
     "pretest": "npm run build",
```

Comment:
If we had a pre-commit hook that checked that `npm run build` was executed (aborting the commit otherwise) e.g. by comparing file timestamps of staged files (hint!) ;-) ,  then we could further speed up tests with:
```
"pretest": "npm run build:fast",
"test:fast": "mocha --timeout 6000 -r ts-node/register/transpile-only \"tests/**/*.spec.ts\"",
```
I think that `"test:fast"` is currently "broken" in that it actually triggers `pretest` (because it calls `npm run test`), which makes it redundant and slower than `npm test`. My suggestion for `test:fast` above is a copy-and-paste of `test`, except that it doesn't trigger `pretest` and doesn't run `build` at all. (It would (only) be useful when the developer makes changes to test code only, not feature code.)

Reply:
I was actually focused on speeding up the CI tests specifically as they run multiple times and having them be less demanding should hopefully help reduce problems 🤞
Speeding up/simplifying the local dev flow specifically is something I'm more focused on in #1589 (particularly around automating it) and is more invasive so I'd rather not bundle it here